model/src/main/kotlin/licenses/LicenseInfoResolver.kt

```diff
@@ -195,6 +202,32 @@ class LicenseInfoResolver(
             ResolvedCopyright(statement, findings.toSet())
         }
     }
+
+    private fun createLicenseFileInfo(id: Identifier): ResolvedLicenseFileInfo {
+        val licenseInfo = resolveLicenseInfo(id)
+        val licenseFiles = mutableListOf<ResolvedLicenseFile>()
+        val matcher = FileMatcher(archiver.patterns)
+
+        licenseInfo.flatMapTo(mutableSetOf()) { resolvedLicense ->
+            resolvedLicense.locations.map { it.provenance }
+        }.forEach { provenance ->
+            val archiveDir = createTempDir(ORT_NAME, "notice").also { it.deleteOnExit() }
+
+            val path = "${id.toPath()}/${provenance.hash()}"
+            if (archiver.unarchive(archiveDir, path)) {
+                archiveDir.walk().forEach { file ->
+                    val relativePath = archiveDir.toPath().relativize(file.toPath()).toString()
```

Comment:
This can again be `val relativePath = file.toRelativeString(archiveDir)`.

Reply:
Done.